### PR TITLE
refactor: use `exactly one` throughout code base

### PR DESCRIPTION
**Pull request by natygyoon**
exactly_one is a neat pattern introduced in [this PR](https://github.com/Unstructured-IO/unstructured/pull/343) that could be leveraged in quite a few other places in the code base. For example, it could be added [here](https://github.com/Unstructured-IO/unstructured/blob/a9152313aae30f7a7a6cf8bddd512afd9d65c3ec/unstructured/file_utils/filetype.py#L131).

### CODE REVIEW

**Comment by ryannikolaidis** [comment]
nit: it looks like this branches from your --metadata-include/exclude branch. because this includes all of those changes, it makes it difficult to review these changes specifically.

File: test_unstructured_ingest/test-ingest-github.sh
Comment (ryannikolaidis) [inline test_unstructured_ingest/test-ingest-github.sh:9]: what was this about? keeping the time down?

File: unstructured/ingest/main.py
Comment (ryannikolaidis) [inline unstructured/ingest/main.py:389]: good catch

**Review by ryannikolaidis — CHANGES_REQUESTED** [review]
The changes that pertain to this PR all look good! I can't approve because you have changes from other PRs mixed in. If you want to wait until those merge and then rebase this or make these change against main, then I can approve.

File: unstructured/ingest/main.py
Comment (ryannikolaidis) [inline unstructured/ingest/main.py:66]: is this meant to be in this PR?

Comment (natygyoon) [reply unstructured/ingest/main.py:66]: Nope, I must have made a mistake while merging main in.

**Review by cragwolfe — APPROVED** [review]
LGTM!

**Review by ryannikolaidis — APPROVED** [review]
Lgtm!

**Pull request merged**
